packages/ubuntu_widgets/lib/src/toggle_button_layout.dart

```diff
@@ -0,0 +1,280 @@
+    return _RenderToggleButton(
+      horizontalSpacing: horizontalSpacing,
+      verticalSpacing: verticalSpacing,
+      textDirection: textDirection,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, _RenderToggleButton renderObject) {
+    renderObject.textDirection = textDirection;
+  }
+}
+
+class _RenderToggleButton extends RenderBox
+    with SlottedContainerRenderObjectMixin<_ToggleButtonSlot> {
+  _RenderToggleButton({
+    required double horizontalSpacing,
+    required double verticalSpacing,
+    required TextDirection textDirection,
+  })  : _horizontalSpacing = horizontalSpacing,
+        _verticalSpacing = verticalSpacing,
+        _textDirection = textDirection;
+
+  RenderBox? get leading => childForSlot(_ToggleButtonSlot.leading);
+  RenderBox? get title => childForSlot(_ToggleButtonSlot.title);
+  RenderBox? get subtitle => childForSlot(_ToggleButtonSlot.subtitle);
+
+  @override
+  Iterable<RenderBox> get children {
+    return <RenderBox>[
+      if (leading != null) leading!,
+      if (title != null) title!,
+      if (subtitle != null) subtitle!
+    ];
+  }
+
+  double get horizontalSpacing => _horizontalSpacing;
+  double _horizontalSpacing;
+  set horizontalSpacing(double value) {
+    if (_horizontalSpacing == value) return;
+    _horizontalSpacing = value;
+    markNeedsLayout();
+  }
+
+  double get verticalSpacing => _verticalSpacing;
+  double _verticalSpacing;
+  set verticalSpacing(double value) {
+    if (_verticalSpacing == value) return;
+    _verticalSpacing = value;
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) return;
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  @override
+  bool get sizedByParent => false;
+
+  static double _minWidth(RenderBox? box, double height) {
+    return box == null ? 0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _maxWidth(RenderBox? box, double height) {
+    return box == null ? 0 : box.getMaxIntrinsicWidth(height);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    return _minWidth(leading, height) +
+        horizontalSpacing +
+        math.max(_minWidth(title, height), _minWidth(subtitle, height));
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _maxWidth(leading, height) +
+        horizontalSpacing +
+        math.max(_maxWidth(title, height), _maxWidth(subtitle, height));
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    final subtitleHeight = subtitle != null
+        ? subtitle!.getMinIntrinsicHeight(width) + verticalSpacing
+        : 0;
+    return title!.getMinIntrinsicHeight(width) + subtitleHeight;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return computeMinIntrinsicHeight(width);
+  }
```

Comment:
Honest question: can there ever be a case in which we may want to let title and subtitle get apart from each other as the window height grows by user resizing it? The method above seems to never allow it.

Reply:
Expanding vertically could be problematic in vertical lists, but we could make it configurable if something like that was needed in the future. I'm glad you asked because this made me realize that the subtitle should wrap. Only the title should ellipsize.